// linguini-app/src/styles/colors.js
const colors = {
  white: '#FFFFFF',
  turquoise: '#40E0D0',
  lightcyan: '#a2fce1',
  orange: '#00e6e6',
  gray: 'rgba(155,155,155,1)',
  transparent: 'transparent',
  red: 'rgba(190, 0, 4, 1)',
  primaryColor: '#00e6e6',
  secondaryColor: '#b2b5b4',

  iconButton: {
    background: 'transparent',
    tint: '#FFFFFF',
    disabled: '#b2b5b4',
  },

  navigationButton: {
    enabled: '#00e6e6',
    disabled: 'rgba(155,155,155,1)',
  },

  audio: {
    recording: 'rgba(190, 0, 4, 1)',
    recordingBackground: '#FFFFFF',
    playButtonBackground: '#FFFFFF',
    startRecordingIcon: '#FFFFFF',
    durationBackground: 'transparent',
  },

  recordVideo: {
    recording: '#000000',
    recordButtonBorder: '#FFFFFF',
    durationText: '#FFFFFF',
    cancelText: '#FFFFFF',
  },

  navHeader: {
    tint: '#f9aa33',
    title: '#000000',
    background: '#4a6572',
  },
  //top
  tabBar: {
    border: '#232f37',
    text: '#4a6572',
    activeBackground: '#232f37',
    inactiveBackground: '#FFFFFF',
  },
  //bottom
  tabNavigator: {
    activeTabIcon: '#00e6e6',
    inactiveTabIcon: '#FFFFFF',
    background: '#232f37',
  },

  empty: {
    title: '#232f37',
    caption: '#FFFFFF',
  },

  border: '#232f37',
};

export default colors;
